src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/IDE/VSSolutionManager.cs

```diff
@@ -281,6 +297,76 @@ public bool IsSolutionOpen
             }
         }
 
+        public async Task<Tuple<bool, bool>> CheckSolutionUIPreConditionsAsync()
```

Comment:
Tuples create unnecessary overhead on the garbage collector (since they are allocated on the heap), i don't really see a strong case for using a tuple here.